src/modules/45-dashboards/pages/home/CreateDashboardForm.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useState } from 'react'
+import { useParams, useHistory } from 'react-router-dom'
+import type { ModalErrorHandlerBinding } from '@wings-software/uicore'
+import { Layout, Container, useToaster } from '@wings-software/uicore'
+import { useCreateDashboard } from 'services/custom-dashboards'
+import type { CreateDashboardResponse } from 'services/custom-dashboards'
+import { useStrings } from 'framework/strings'
+import routes from '@common/RouteDefinitions'
+import type { DashboardPathProps } from '@common/interfaces/RouteInterfaces'
+import DashboardForm, { DashboardFormRequestProps } from './DashboardForm'
+import css from './HomePage.module.scss'
+
+interface CreateDashboardFormProps {
+  hideModal: () => void
+}
+
+const CreateDashboardForm: React.FC<CreateDashboardFormProps> = ({ hideModal }) => {
+  const { getString } = useStrings()
+  const { accountId, folderId } = useParams<DashboardPathProps>()
+  const [modalErrorHandler, setModalErrorHandler] = useState<ModalErrorHandlerBinding>()
+  const history = useHistory()
+  const { showSuccess } = useToaster()
+
+  const { mutate: createDashboard, loading } = useCreateDashboard({
+    queryParams: { accountId: accountId }
+  })
+
+  const onSuccess = (data: CreateDashboardResponse): void => {
+    if (data?.resource) {
+      showSuccess(getString('dashboards.createModal.success'))
+      hideModal()
+      history.push({
+        pathname: routes.toViewCustomDashboard({
+          viewId: data.resource.toString(),
+          accountId: accountId,
```

Comment:
1000% optional, but you can shorthand this.
```suggestion
          accountId,
```